engine/common/synchronization/engine.go

```diff
@@ -307,15 +306,9 @@ func (e *Engine) onBlockResponse(originID flow.Identifier, res *messages.BlockRe
 			e.log.Debug().Uint64("height", block.Header.Height).Msg("block handler rejected")
 			continue
 		}
-		synced := &events.SyncedBlock{
```

Comment:
With this change, we will be forwarding the `messages.BlockResponse` to compliance engine, instead of converting to `SyncedBlock`. 

We've changed the follower engine to handle `messages.BlockResponse`, but we haven't changed the [consensus compliance engine](https://github.com/onflow/flow-go/blob/master/engine/consensus/compliance/engine.go#L105) and [collection compliance engine](https://github.com/onflow/flow-go/blob/master/engine/collection/compliance/engine.go#L120) to process `messages.BlockResponse`, which means this will break forward syncing for them.

In order to fix it, we need to add a new FIFOQueue as `pendingRangeResponses` just like the [pendingBlocks](https://github.com/onflow/flow-go/blob/master/engine/collection/compliance/engine.go#L90), and add a [Pattern](https://github.com/onflow/flow-go/blob/master/engine/collection/compliance/engine.go#L108) to the message handler to handle the RangeResponse. 

Then when processing the RangeResponse, we will be able to pass the inRangeResponse parameter to the OnBlockProposal call.